Draft-Accepted/RFCNNNN-PathUpdateForPackageInstall.md

```diff
@@ -0,0 +1,189 @@
+---
+RFC: RFC<four digit unique incrementing number assigned by Committee, this shall be left blank by the author>
+Author: Dongbo Wang
+Status: Draft
+SupercededBy:
+Version: 1.0
+Area: Core
+Comments Due: 5/30/2025
+Plan to implement: Yes
+---
+
+# Path update for package installation
+
+On Windows, when a user installs a tool using package managers like `WinGet` and `Chocolatey` in PowerShell,
+the tool is often added to the system or user PATH environment variable. However, changes to PATH made during
+the installation are not automatically propagated to the running PowerShell session. As a result, users frequently
+need to start a new PowerShell session to recognize and use the newly installed command-line tools.
+
+This behavior can be confusing and disrupts the workflow, especially in automation scenarios or when users expect
+tools to be immediately available post-installation. Addressing this limitation would improve usability, align with
+user expectations, and reduce friction in both interactive and scripted environments.
+
+## Motivation
+
+As a PowerShell user, I would like tools installed by package managers like `WinGet` or `Chocolatey` to be useable
+in my current session so that I can use the new tool without interrupting my existing work in the current session.
+
+## Solution in CMD
+
+CMD suffered with the same problem and the Windows Terminal team already [made a fix](https://microsoft.visualstudio.com/OS/_git/os.2020/pullrequest/10131240)
+to mitigate the issue:
+
+> With this change, CMD will detect the execution of `winget.exe` and `choco.exe`,
+> as well as other package managers (the list is configurable in the registry given HKLM write permission).
+> If we are launching a package manager, we will save the current user and system `PATH` and diff them
+> after the package manager exits.
+>
+> To make this safe, there are a couple caveats:
+>
+> 1. We will only detect changes to the BEGINNING or END of either `PATH` (user/system).
+> 2. We will only ever **append** to CMD's current `PATH`.
+> 3. This feature can be disabled by removing the list of package managers at `HKLM\SOFTWARE\Microsoft\Command Processor\KnownPackageManagers`.
```

Comment:
Have you asked WT team why they need to have this list? Why not update the PATH every time (it's not that expensive)?

Reply:
The idea is to make only necessary changes to the existing behavior.